onadata/apps/api/viewsets/user_profile_viewset.py

```diff
@@ -200,3 +213,84 @@ def monthly_submissions(self, request, *args, **kwargs):
 
         serializer = MonthlySubmissionsSerializer(instance_count, many=True)
         return Response(serializer.data[0])
+
+    @action(detail=False)
+    def verify_email(self, request, *args, **kwargs):
+        verified_key_text = getattr(settings, "VERIFIED_KEY_TEXT", None)
+
+        if not verified_key_text:
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        redirect_url = request.query_params.get('redirect_url')
+        verification_key = request.query_params.get('verification_key')
+        response_message = _("Missing or invalid verification key")
+        if verification_key:
+            try:
+                rp = RegistrationProfile.objects.get(
+                    activation_key=verification_key
+                )
+            except RegistrationProfile.DoesNotExist:
+                pass
+            else:
+                rp.activation_key = verified_key_text
```

Comment:
Why this assignment?

Reply:
As explained above, it just assigns `ALREADY_ACTIVATED'` to `activation_key` field.